### PR TITLE
Add qhttp -- lightweight C++ HTTP server via boost::asio

### PR DESCRIPTION
**Pull request by fritzm**
This is the same code as previously had a preliminary review, but now with:
- a README.md (including usage examples)
- some code comments in the headers
- fairly complete unit tests

### CODE REVIEW

File: qhttp/AjaxEndpoint.cc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include "Server.hpp"
+
+namespace lsst {
+namespace qserv {
+namespace qhttp {
+
+
+AjaxEndpoint::Ptr AjaxEndpoint::add(Server& server, std::string const& path)
+{
+    auto aep = std::shared_ptr<AjaxEndpoint>(new AjaxEndpoint);
```
Comment (jgates108) [inline qhttp/AjaxEndpoint.cc:21]: std::make_shared?

Comment (fritzm) [inline qhttp/AjaxEndpoint.cc:21]: Probably should have a private ctor instead -- the intention is that they are only created by add().

Comment (jgates108) [inline qhttp/AjaxEndpoint.cc:21]: Private constructor sounds like the better choice.

**Comment by jgates108** [comment]
It looks good. The code has lots of fine details that can be tricky, so I'm glad to see that the unit test coverage looks thorough. The coding style is a bit different than the rest of Qserv.

**Pull request merged**